utils/calendrical_calculations/src/islamic.rs

```diff
@@ -0,0 +1,271 @@
+use crate::astronomy::*;
+use crate::helpers::{self, div_rem_euclid, div_rem_euclid64, div_rem_euclid_f64, next};
+use crate::rata_die::{Moment, RataDie};
+
+// Different islamic calendars use different epochs (Thursday vs Friday) due to disagreement on the exact date of Mohammed's migration to Mecca.
+/// Lisp code reference: <https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2066>
+const FIXED_ISLAMIC_EPOCH_FRIDAY: RataDie = crate::julian::fixed_from_julian(622, 7, 16);
+const FIXED_ISLAMIC_EPOCH_THURSDAY: RataDie = crate::julian::fixed_from_julian(622, 7, 15);
+
+/// Lisp code reference: <https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L6898>
+const CAIRO: Location = Location {
+    latitude: 30.1,
+    longitude: 31.3,
+    elevation: 200.0,
+    zone: (1_f64 / 12_f64),
+};
+
+// "Fixed" is a day count representation of calendars staring from Jan 1st of year 1 of the Georgian Calendar.
+// The fixed date algorithms are from
+// Dershowitz, Nachum, and Edward M. Reingold. _Calendrical calculations_. Cambridge University Press, 2008.
+/// Lisp code reference: <https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L6904>
```

Comment:
question: Which comments should be Rustdoc vs. "regular" non-doc?

The pattern that I observe in this module is that the "Lisp code reference: ..." comments are Rustdoc, but if there is an explanatory comment, it doesn't make it into the Rust docs. But changing the explanatory comments to be included in Rustdocs seems preferable.

Reply:
a lot of this is copied boilerplate and should be cleaned up